lessons/index.html

```diff
@@ -3,310 +3,328 @@
-  <li class="lesson"><a href="module-4/intro-to-sql.html">Intro to SQL</a></li>
-  <li class="lesson"><a href="module-4/knex-postgres.html">Express with Knex and Postgres</a></li>
-</ul>
-
-
-<h3>Security &amp; Authentication</h3>
-
-<ul>
-  <li class="lesson"><a href="module-4/client-side-security.html">Intro to Client Side Security</a></li>
-  <li class="lesson"><a href="module-4/http-vs-https.html">HTTP vs HTTPS</a></li>
-  <li class="lesson"><a href="module-4/jwts/index.html">Intro to JWTS</a></li>
-  <li class="lesson"><a href="module-4/oauth/index.html">Intro to OAuth</a></li>
-</ul>
-
-<h3>Build Tools &amp; Workflow</h3>
-
-<ul>
-  <li class="lesson"><a href="module-4/environments.html">Environments &amp; Their Variables</a></li>
-  <li class="lesson"><a href="module-4/advanced-git-workflow.html">Advanced git Workflow</a></li>
-  <li class="lesson"><a href="module-4/continuous-integration.html">Continuous Integration with CircleCI</a></li>
-  <li class="lesson"><a href="module-4/better-error-workflow.html">Better Error Workflow</a></li>
-</ul>
-
-<h3>PWAs, Service Workers &amp; Offlining Applications</h3>
-
-<ul>
-  <li class="lesson"><a href="module-4/pwas/index.html">Intro to PWAs</a></li>
-  <li class="lesson"><a href="module-4/pwas/service-workers.html">Intro to Service Workers</a> | <a href="module-4/service-worker-slides.html">Slides</a></li>
-  <li class="lesson"><a href="module-4/indexedDB.html">Persisting Data with IndexedDB</a> | <a href="module-4/persisting-with-indexeddb.html">Slides</a></li>
-  <li class="lesson"><a href="module-4/background-sync.html">Background Sync</a> | <a href="module-4/sync-push.html">Slides</a></li>
-  <li class="lesson"><a href="module-4/react-native-intro.html">React Native Intro</a></li>
-</ul>
-
-<h3>General Web</h3>
-<h4>Performance</h4>
-<ul>
-  <li class="lesson"><a href="module-4/intro-to-performance.html">Performance</a></li>
-</ul>
-
-
-<h4>Accessibility</h4>
-<ul>
-  <li class="lesson"><a href="module-4/localization.html">Localization</a></li>
-</ul>
-
-<h3>Other</h3>
-
-<ul>
-  <li class="lesson"><a href="module-4/practice-interview-questions/interview-questions.html">Interview Questions</a></li>
-  <li class="lesson"><a href="module-4/web-scraping-workshop.html">Web Scraping Workshop</a></li>
-  <li class="lesson"><a href="module-4/get-low-level.html">Get Low-Level: Understanding Specs and Standards Bodies</a></li>
-  <li class="lesson"><a href="module-4/sorting-algorithms.html">Sorting Algorithms</a></li>
-  <li class="lesson"><a href="module-4/search-algorithms.html">Search Algorithms</a></li>
-</ul>
\ No newline at end of file
+<section>
+  <button class="module-selector" data-module="1">Module 1</button>
+  <button class="module-selector" data-module="2">Module 2</button>
+  <button class="module-selector" data-module="3">Module 3</button>
+  <button class="module-selector" data-module="4">Module 4</button>
+  <button class="module-selector" data-module="0">Show All</button>
```

Comment:
A PERFECT USE OF DATA ATTRIBUTES. I always have a hard time finding an example of why these are useful AF for students.